internal/pkg/source/source.go

```diff
@@ -162,11 +162,35 @@ func (s *Source) referrersToVisit(n ssa.Node, maxInstrReached map[*ssa.BasicBloc
 			continue
 		}
 
+		if e, ok := r.(*ssa.Extract); ok && shouldNotVisitExtract(n, e) {
+			continue
+		}
+
 		referrers = append(referrers, r)
 	}
 	return referrers
 }
 
+// shouldNotVisitExtract determines if the extract is extracting a boolean from
+// a comma-ok map lookup or chan recv, in which case we should not visit it,
+// because a boolean cannot meaningfully be tainted.
+func shouldNotVisitExtract(n ssa.Node, e *ssa.Extract) bool {
+	// the boolean in a comma-ok extract is always at the 1st index
+	if e.Index != 1 {
+		return false
+	}
+	switch n.(type) {
+	// Lookup is for a map lookup, e.g.
+	// maybeValue, ok := myMap[myKey]
+	// UnOp is for a chan recv, e.g.
+	// maybeValue, ok := <-myChan
+	case *ssa.Lookup, *ssa.UnOp:
```

Comment:
The following `Instruction` types need to be unpacked by an `Extract`.

* `ssa.TypeAssert` appears to be the only other type that stores a `CommaOk bool` field.  Should we add that to this set?

* `Call` and `Return` bundle return types for an arbitrary function call.  We shouldn't make a value judgement on any particular element based on return position.  We might include them in the case switch if we were being explicit in our enumeration.  Either way.

* `ssa.Next` reutrns a 3-tuple, where we should also not propagate to the `bool`.  This could be a part of a future PR / deeper future examination.

* `ssa.Select` contains a much more complex `n+2` tuple, but I'm not sure we have any `select` tests yet.  That probably puts it as a candidate for a future PR.

Reply:
Good point on `TypeAssert`, I figured it was alright because we're already avoiding propagating *from* an `Extract` to the instruction it is extracting from. So that covers e.g. `s, ok := (interface{}).(Source)`. However, if the interface value is tainted, then a different logic applies. I was missing a test case.